SU2_CFD/src/numerics/turbulent/turb_sources.cpp

```diff
@@ -0,0 +1,1105 @@
+/*!
+ * \file turb_sources.cpp
+ * \brief Implementation of numerics classes for integration of
+ *        turbulence source-terms.
+ * \author F. Palacios, T. Economon
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../include/numerics/turbulent/turb_sources.hpp"
+
+CSourceBase_TurbSA::CSourceBase_TurbSA(unsigned short val_nDim,
+                                       unsigned short val_nVar,
+                                       const CConfig* config) :
+  CNumerics(val_nDim, val_nVar, config),
+  incompressible(config->GetKind_Regime() == INCOMPRESSIBLE),
+  rotating_frame(config->GetRotating_Frame())
+{
+  /*--- Spalart-Allmaras closure constants ---*/
+
+  cv1_3 = pow(7.1, 3.0);
+  k2    = pow(0.41, 2.0);
+  cb1   = 0.1355;
+  cw2   = 0.3;
+  ct3   = 1.2;
+  ct4   = 0.5;
+  cw3_6 = pow(2.0, 6.0);
+  sigma = 2./3.;
+  cb2   = 0.622;
+  cb2_sigma = cb2/sigma;
+  cw1 = cb1/k2+(1.0+cb2)/sigma;
+
+  /*--- Setup the Jacobian pointer, we need to return su2double** but
+   *    we know the Jacobian is 1x1 so we use this "trick" to avoid
+   *    having to dynamically allocate. ---*/
+
+  Jacobian_i = &Jacobian_Buffer;
+
+}
```

Comment:
I have also done some cleanup, for example by inserting some intermediate classes to at least centralise the definition of hardcoded values.